inference-engine/thirdparty/clDNN/src/graph_optimizer/compile_graph.cpp

```diff
@@ -10,17 +10,64 @@
 #include "program_node.h"
 #include "cldnn/runtime/engine.hpp"
 #include "runtime/cldnn_itt.hpp"
+#include <iostream>
+#include <cmath>
+#include <iomanip>
+
+#define CLDNN_THREADING_SEQ 0
+#define CLDNN_THREADING_TBB 1
+
+#if (CLDNN_THREADING == CLDNN_THREADING_TBB)
+#include <tbb/parallel_for.h>
+#include <tbb/blocked_range.h>
+#endif
 
 using namespace cldnn;
 
+inline std::string convert_fixed_length_num(const size_t num, const int fixed_length) {
+    std::ostringstream out_strstr;
+    out_strstr << std::setfill('0') << std::setw(fixed_length) << num;
+    return out_strstr.str();
+}
+
 void compile_graph::run(program_impl& p) {
     OV_ITT_SCOPED_TASK(itt::domains::CLDNN, "CLDNN::pass::CompileGraph");
+#if (CLDNN_THREADING == CLDNN_THREADING_TBB)
+    for (auto& node : p.get_processing_order()) {
+        if (!node->is_type<data>()) {
+            node->get_output_layout();
```

Comment:
When some node and its parent node are running on different threads, 
While the child node called get_output_layout() and before calling get_default_params() in choose_impl(), the parent node calls get_output_layout() and invalidates all child nodes.

In this case, an exception occurs while the child node performs get_output_layout() inside get_default_params().
To fix this issue, modified codes to run get_output_layout() sequentially before choose_impl parallel processing.

I also think get_output_layout() doesn't change anything at first. However, I found that It sometimes invalidates users because its output layout are changed.

Reply:
I got it, thanks for explaining.